apps/rush-lib/src/logic/deploy/DeployArchiver.ts

```diff
@@ -0,0 +1,97 @@
+import * as JSZip from 'jszip';
+import * as path from 'path';
+import { FileSystem, FileSystemStats, AlreadyExistsBehavior } from '@rushstack/node-core-library';
+import { IDeployState } from './DeployManager';
+
+// JSZip is dependant on Blob being declared.
+declare global {
+  // eslint-disable-next-line
+  type Blob = any;
+}
+export class DeployArchiver {
+  public static async createArchive(deployState: IDeployState): Promise<void> {
+    if (deployState.scenarioConfiguration.json.postCopySourceFolder !== undefined) {
+      const sourceFolderPath: string = FileSystem.getRealPath(
+        deployState.scenarioConfiguration.json.postCopySourceFolder
```

Comment:
This is resolved relative to the current working directory, which causes the copy to fail if you are not in the repository root.  You can use something like `path.resolve(this._rushConfiguration.rushJsonFolder, deployState.scenarioConfiguration.json.postCopySourceFolder)` to fix that.

Reply:
The failure itself looked like this:
```
The operation completed successfully.
(node:7752) UnhandledPromiseRejectionWarning: Error: File does not exist: C:\Git\rushstack2\apps\rush-lib\common
ENOENT: no such file or directory, lstat 'C:\Git\rushstack2\apps\rush-lib\common'
    at Object.realpathSync (fs.js:1615:7)
    at C:\Git\rushstack2\libraries\node-core-library\lib\FileSystem.js:723:24
```
...which implies that a promise chain is not being terminated correctly. 

Lastly, if the `postCopySourceFolder` doesn't exist, it probably shouldn't be a hard error. We could log a warning maybe.

Reply:
Otherwise the command did work correctly when I tried it!  💪🙂